### ISSUE TITLE
Url has never been harvested:http://null/...

### ISSUE THREAD

**Issue by mbreemhaar** [netarchivesuite/solrwayback#231]
I have a problem with [this WARC file](https://drive.google.com/file/d/1Jv871Tjls752AAEf_RAXzk-kVaNmv8l0/view?usp=sharing) of www.roodjongeren.nl.

If I click any link on the archived website, it will show me that "Url has never been harvested:http://null/...".

This seems strange, because if I look at the source code of the page, the link does actually refer to the correct page, so not "http://null/...".

The WARC file is available at https://drive.google.com/file/d/1Jv871Tjls752AAEf_RAXzk-kVaNmv8l0/view?usp=sharing

Can you help me with this problem? Thank you.

**Comment by tokee** [comment]
There seems to be some quite strange interaction with the page's JavaScript. I did not get to the bottom of it, but here's my notes so that @thomasegense can follow up when he gets back:

I reproduced by
```
unzip ~/Downloads/solrwayback_package_4.3.0.zip
cd solrwayback_package_4.3.0
cp properties/*.properties ~/
apache-tomcat-8.5.60/bin/startup.sh
cd indexing/
cp ../roodjongeren.warc warcs1/
./warc-indexer.sh warcs1
```

After that http://localhost:8080/solrwayback/services/web/20220810131131/https:/roodjongeren.nl shows the ROOD homepage. 

1. Hovering over an article link shows a SolrWayback-URL, e.g. http://localhost:8080/solrwayback/services/web/20220810131440/https:/roodjongeren.nl/post/Cultuurmarxisme-in-Oost-Groningen
2. Clicking such a link results in the null-problem described and the URL http://localhost:8080/solrwayback/services/webProxy/20220810131131/http://null/post/Cultuurmarxisme-in-Oost-Groningen (notice that `roodjongeren.nl` is replaced with `null`.
4. Using Firefox and CTLR-clicking opens a new tab, where the correct URL (the one with the domain name intact) is played back. The URL is harvested and playback works fine
5. Copy-pasting the hover-URL from 1) also works

There is a JavaScript-event to a function, but the debugger is not of much help as it says the function is empty: `function Vr(){}`. A look in the raw content from the WARC confirms this, buried in a lot of obfuscated/minified JavaScript: `}0!==o.length&&e.push({event:t,listeners:o})}function Vr(){}var jr=null,Br=null;`.

Checking network connections in the browser, it _seems_ that clicking a link makes a request to  http://localhost:8080/post/Cultuurmarxisme-in-Oost-Groningen instead of a domain-qualified URL. However this is with a proper referer, so I would guess that it should work anyway?

**Comment by thomasegense** [comment]
@mbreemhaar 

Try this patch - download ROOT.war here:
https://drive.google.com/file/d/1j1qnygdvsVJ-ZRzG3Ap-ZOFzuEO47Rkd/view?usp=sharing
Replace the file in the 'apache-tomcat-8.5.60/webapps' and restart the tomcat.

Please confirm if it works for you and I will merge the branch  into solrwaybackrootproxy-master.  Thanks for the bug-report. There are still many tricky playback things that needs fixed. The other bug you reported #230 is much more serious and finding it was helpful since it is very hard to track down and can happen much more often. I hope we have a fix soon for that.

**Comment by thomasegense** [comment]
@mbreemhaar  Hi. 
Had time to try the fix?

**Comment by mbreemhaar** [comment]
Sorry for the late response. Yes, the problem seems to be resolved with this patch. Thank you very much!

**Comment by thomasegense** [comment]
Thanks. Next bundle release will contain this fix. Next release 4.3.1 will not be released before a few more issues has been fixed such as the #230 you reported.
